parachain/src/wasm_executor/validation_host.rs

```diff
@@ -242,12 +259,17 @@ impl ValidationHost {
 			}
 		}
 		let memory = Self::create_memory()?;
-		let self_path = env::current_exe()?;
-		debug!("Starting worker at {:?}", self_path);
-		let mut args = if test_mode { WORKER_ARGS_TEST.to_vec() } else { WORKER_ARGS.to_vec() };
-		args.push(memory.get_os_path());
-		let worker = process::Command::new(self_path)
+		let (cmd, args) = match execution_mode {
+			ValidationExecutionMode::Local => (
+				env::current_exe()?,
+				WORKER_ARGS.iter().map(|x| x.to_string()).collect()
+			),
+			ValidationExecutionMode::Remote { binary, args } => (binary, args),
+		};
```

Comment:
Wait. You clear did not understand what Local and Remote mean here.

Local => Run the verification inside the current process
Remote => Use the given external command and args to start a worker.

Reply:
I actually would prefer an enum option to use default args and executable name. Defaults should work in all cases other than testing, and requiring the user to always specify executable/args would be bad API design. It is very much an internal implementation details and it only leaks away because tests require special handling.

Reply:
What you mean by enum option?

Reply:
An enum value that does not require setting executable name for out-of-process validation. E.g. `RemoteDefault` or simply `Remote`.  Basically something like this:
```rust
pub enum ValidationExecutionMode {
	InProcess,
        ExternalProcessSelfHost, 
	ExternalProcessCustomHost {
		binary: PathBuf,
		args: Vec<String>,
	},
```

Reply:
Okay. Sounds reasonable.